components/net/http_cache.rs

```diff
@@ -519,7 +554,23 @@ fn handle_range_request(
                 } else {
                     continue;
                 };
-                if (total - res_beginning) > (offset - 1) && (total - res_end) < offset + 1 {
+                let (total_minus_res_beginning, total_minus_res_end) = {
```

Comment:
If we check that `total >= res_beginning` and `total >= res_end` then we don't need these checked operations.